### PR TITLE
Organisation members page to show all users

### LINKED ISSUE #19511: New org `/members` list and limit members shown on organization page
## Is this feature related to a problem? Please describe.

* Users are not capped on organization pages (e.g. https://dev.to/aws-builders).
* As a user visiting an organization page with a large number of members, I want to see an ordered, tidy list.

## Describe the solution you’d like

1. New organization `/members` page:

- 1a
  - All organizations will now have a `/members` page (e.g. `/https://dev.to/aws-builders/members`), which will list all of the members in the organization
  - The users should be ordered by the number of badges the users have, in descending order.
  - Each listed user should display have a link to their profile
  - A page/list design has been created for this, which in future will be used for all future lists of users.

- 1b
  - Allow users to follow other users easily on this listing, by showing a 'Follow' button

2. Truncate the list of users on the organization page
  - The list of users, shown in the left sidebar on the organization page, will cap at 50.
  - When the cap is placed, a link to the `/members` page will be shown (e.g. "See all members").

**Designs** - [Figma link](https://www.figma.com/file/9UfSjNYJs745w3Q8JCUqsL/Small-Wins---%E2%80%A8Core-Experience-%E2%9A%A1%EF%B8%8F?type=design&node-id=2080-9066&t=2mctDCv5ltqWSjth-4)

Final

### Deliverable 1a
- [ ] New organization `/members` page
  - [ ]  Ordered by user's number of badges, in descending order
  - [ ] Able to access the user's profile, via a link
### Deliverable 1b
- [ ] Users able to follow others users from this listing

### Deliverable 2
- [ ] Organization page capped at 50 members, with link to `/members` page

## Additional Context

- Be sure to use the cached value of the user's badges that is pre-calculated (`users.badge_achievements_count`).
This work will coincide with the work to re-order the profile images shown on the organization and tag pages (https://github.com/forem/forem/issues/19509).
- In future phases, we would want to show the 'profile hover' functionality in this space, which allows users to hover over a linked profile or profile image to see the details of the user's profile and follow the user.

### PR DESCRIPTION
**Pull request by rt4914**
## What type of PR is this? (check all applicable)

- [x] Feature

## Description

This PR introduces new `/members` page on organisation page which shows all users of that org.

There are two major changes:
1. Create new `/members` page on org page
2. Update org page sidebar to limit the users to `50` and show `See All Members` button at bottom.

## Related Tickets & Documents

- Closes #19511

## QA Instructions, Screenshots, Recordings

Designs - [Figma link](https://www.figma.com/file/9UfSjNYJs745w3Q8JCUqsL/Small-Wins---%E2%80%A8Core-Experience-%E2%9A%A1%EF%B8%8F?type=design&node-id=2080-9066&t=2mctDCv5ltqWSjth-4)

**How to test?**
1. First make a change in `app/controllers/stories_controller.rb`, change `user_limit` from 50 to 2 or 3 so that you can easily test it.
2. Now visit any organisation page which has more than the limit and notice `See All Members` button, click on it to visit `/members` page.

**Output**

| Light  | Dark |
| ------------- | ------------- |
| Screenshot 2023-06-16 at 3 05 25 PM | Screenshot 2023-06-16 at 3 05 10 PM |
| Screenshot 2023-06-16 at 3 05 38 PM | Screenshot 2023-06-16 at 3 05 01 PM |

Screenshot 2023-06-23 at 1 17 35 PM
Screenshot 2023-06-23 at 1 17 26 PM
Screenshot 2023-06-23 at 1 17 19 PM
Screenshot 2023-06-23 at 1 17 11 PM Screenshot 2023-06-23 at 1 17 03 PM Screenshot 2023-06-23 at 1 16 55 PM

### UI accessibility checklist
_If your PR includes UI changes, please utilize this checklist:_
- [x] Semantic HTML implemented?
- [x] Keyboard operability supported?
- [x] Checked with [axe DevTools](https://www.deque.com/axe/) and addressed `Critical` and `Serious` issues?
- [x] Color contrast tested?

Screenshot 2023-06-16 at 5 24 29 PM

_For more info, check out the
[Forem Accessibility Docs](https://developers.forem.com/frontend/accessibility)._

## Added/updated tests?

- [x] Yes

### CODE REVIEW

**Comment by rt4914** [comment]
> @rt4914
> 
> 1. I couldn’t check on Uffizzi. But I hope it's the same as [this CTA component](https://storybook.forem.com/?path=/story/components-navigation-ctas--default)
> 2. [According to issue](https://github.com/forem/forem/issues/19511), I think it's only when there are more than 50 members.   But the user should be able to access the `/members` page manually. @PhilipHow please confirm if it's the expected behavior or not.
> 3. This is perfect. We can also attach a link to the name and username as well. All should redirect to the profile. In the future, we can have a profile dropdown on name hover.
> 4. Overall looks great. My only suggestion is to keep the card height fixed. Since we are expanding long names to two lines, it's increasing the card height as well. Instead, we can truncate _(E.g. Anuj ABC Bhavs...)_ the name in the case when the name is overflowing.

@anujbhavsar96 I have applied these design changes:

Screenshot 2023-06-20 at 6 09 15 PM
Screenshot 2023-06-20 at 6 11 30 PM

**Comment by PhilipHow** [comment]
@rt4914 yeah only show the link when there is more than 50 members.

File: app/views/organizations/_sidebar.html.erb [hunk 1]
```diff
@@ -10,14 +10,21 @@
...
           <div class="org-sidebar-widget-user-pic">
             <a href="/<%= user.username %>">
               <img src="<%= user.profile_image_url_for(length: 90) %>" alt="<%= user.username %> profile image" loading="lazy" />
             </a>
           </div>
         <% end %>
       </div>
+      <% if @organization_users.count > @user_limit %>
+        <div class="see-all-members-container">
+          <button class="c-cta" onclick="location.href='/<%= @organization.slug %>/members'">
```
[the remark this answers is not in the document]
Comment (rt4914) [reply app/views/organizations/_sidebar.html.erb]: I have applied this changed.
Removed the div and also changed button to `a` tag.

File: app/assets/stylesheets/widgets.scss [hunk 2]
```diff
@@ -284,6 +284,18 @@
   overflow: hidden;
   display: block;
 
+  .see-all-members-container {
```
[the remark this answers is not in the document]
Comment (rt4914) [reply app/assets/stylesheets/widgets.scss]: I have applied changes to this. I have kept minimal properties which are required to provide correct UI.

**Comment by rt4914** [comment]
@anujbhavsar96 Removed fixed height to fix the padding issue.

Screenshot 2023-06-23 at 12 41 54 PM

**Review by anujbhavsar96 — APPROVED** [review]
Looks perfect 🌟

**Review by benhalpern — APPROVED** [review]

**Comment by rt4914** [comment]
Hi @lboogie-ftf Looking for one more eng-review in this PR. Thanks.

**Comment by rt4914** [comment]
Thanks @mirie 
Will delete that comment.

**Comment by rt4914** [comment]
Assigned this PR to @jaw6 and @filleduchaos for one more review. Thanks.

**Review by jaw6 — APPROVED** [review]

**Pull request merged**